### PR TITLE
feat(docs): update radio page to use new layout

### PR DESCRIPTION
**Pull request by MariaJose**
## What?
Update radio page to use new layout.

## Why?
Provide better experience to the user when reading the docs.

## Screenshots

Screen Shot 2022-02-22 at 1 57 40 PM

### CODE REVIEW

File: packages/docs/PropTables/shared.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const messagingLinkItemProps: Prop[] = [
     types: 'boolean',
     description: (
       <>
-        Shows an external icons when the <Code primary>external</Code> flag is set and target="_blank".
+        Shows an external icon when the <Code primary>external</Code> flag is set and target="_blank".
```
Comment (MariaJose) [inline packages/docs/PropTables/shared.tsx:70]: Should we update `external flag` to `external prop`?

File: packages/docs/pages/radio.tsx [hunk 2]
```diff
@@ -1,151 +1,219 @@
...
+                </Fragment>
+              ),
+            },
+            {
+              id: 'grouping',
+              title: 'Grouping',
+              render: () => (
+                <Fragment key="grouping">
+                  <Text>
+                    In order to group radio controls, use the <Code>Fieldset</Code> component to separate the controls.
```
Comment (MariaJose) [inline packages/docs/pages/radio.tsx:68]: Tried to find the `Fieldset` page, but I didn't find it, Is that expected?
Should  we remove the `<Code/>` tag?

File: packages/docs/PropTables/shared.tsx (hunk 1 above)
Comment (jorgemoya) [reply packages/docs/PropTables/shared.tsx:70]: Sounds like a good idea to me

File: packages/docs/pages/radio.tsx (hunk 2 above)
Comment (jorgemoya) [reply packages/docs/pages/radio.tsx:68]: I guess we're referencing the html element. I would remove the `Code` tag.

Comment (chanceaclark) [reply packages/docs/pages/radio.tsx:68]: We have a component for [`Fieldset`](https://github.com/bigcommerce/big-design/blob/master/packages/big-design/src/components/Fieldset/Fieldset.tsx) but we don't have a dedicated page for it. It's located on the `Forms` page props.

Comment (jorgemoya) [reply packages/docs/pages/radio.tsx:68]: We group `Fieldset` in the `Form` component. I think it's fine to leave as is.

**Review by chanceaclark — APPROVED** [review]
Besides the `flag` -> `prop` change, lgtm

**Review by jorgemoya — APPROVED** [review]

File: packages/docs/pages/radio.tsx [hunk 3]
```diff
@@ -1,151 +1,219 @@
...
-              {/* jsx-to-string:start */}
-              {function Example() {
-                const [selected, setSelected] = useState('');
+  return (
+    <>
+      <H1>Radio</H1>
 
-                const handleChange = (event) => setSelected(event.target.value);
+      <Panel header="Overview" headerId="overview">
+        <Text>Radio buttons let users select an option from a list of two or more.</Text>
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:13]:
```suggestion
        <Text>Radio buttons let users select an option from a list of two or more items.</Text>
```

File: packages/docs/pages/radio.tsx [hunk 4]
```diff
@@ -1,151 +1,219 @@
...
+        <ContentRoutingTabs
+          id="implementation"
+          routes={[
+            {
+              id: 'basic',
+              title: 'Basic',
+              render: () => (
+                <Fragment key="basic">
+                  <Text>
+                    <Code primary>Radios</Code> are group of items which a single option can be selected.
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:32]:
```suggestion
                    A <Code primary>Radio</Code> is a group of items from which a single option can be selected.
```

File: packages/docs/pages/radio.tsx [hunk 5]
```diff
@@ -1,151 +1,219 @@
...
+          ]}
+        />
+      </Panel>
 
-      <PageNavigation items={items} />
+      <Panel header="Do's and Don'ts" headerId="guidelines">
+        <GuidelinesTable
+          recommended={[
+            <>Group related radio buttons under input headings.</>,
+            <>Radio buttons should always have a default selected option.</>,
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:206]:
```suggestion
            <>Include a default selected option with the radio buttons.</>,
```

File: packages/docs/pages/radio.tsx [hunk 6]
```diff
@@ -1,151 +1,219 @@
...
+            <>Radio buttons should always be laid out vertically.</>,
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:207]:
```suggestion
            <>Lay radio buttons vertically.</>,
```

File: packages/docs/pages/radio.tsx [hunk 7]
```diff
@@ -1,151 +1,219 @@
...
+          ]}
+          discouraged={[
+            <>Don’t use radio buttons for long lists of short items - use a select input instead.</>,
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:210]:
```suggestion
            <>Don’t use radio buttons for long lists of short items. Use a select input instead.</>,
```

File: packages/docs/pages/radio.tsx [hunk 8]
```diff
@@ -1,151 +1,219 @@
...
+            <>
+              A set of radio buttons should not have a state of being “unselected” - there must always be a selection.
```
Comment (bc-andreadao) [inline packages/docs/pages/radio.tsx:212]:
```suggestion
              A set of radio buttons should not have a state of being “unselected.” There must always be a selection.
```

**Comment by MariaJose** [comment]
♻️

**Review by jorgemoya — APPROVED** [review]

**Review by bc-andreadao — APPROVED** [review]

**Pull request merged**